src/commands.rs

```diff
@@ -4948,12 +4948,31 @@ fn cmd_git_push(
             old_heads.extend(old_head.adds());
         }
     }
+    if old_heads.is_empty() {
+        old_heads.push(repo.store().root_commit_id().clone());
+    }
     for index_entry in repo.index().walk_revs(&new_heads, &old_heads) {
         let commit = repo.store().get_commit(&index_entry.commit_id())?;
-        if commit.tree().has_conflict() {
+        let reason = if commit.description().is_empty() {
+            Some("it has no description")
```

Comment:
Uncertain: should we make a list of possible reasons and emit them all? This way users won't need to run it multiple times solving one problem each time. On the other hand, getting a message like: `Won't push commit {} since it has no description, it has no author set, it has no committer set, and it has conflicts.` might be a bit demoralizing.

Reply:
I did that but I merged the "no author" and "no committer" cases into one since they should be fixed at the same time now that we reset the author field together with the committer field once the user has set configured their name/email.